src/librustdoc/clean/utils.rs

```diff
@@ -141,19 +141,21 @@ fn external_generic_args(
     }
 }
 
-// trait_did should be set to a trait's DefId if called on a TraitRef, in order to sugar
-// from Fn<(A, B,), C> to Fn(A, B) -> C
+/// trait_did should be set to a trait's DefId if called on a TraitRef, in order to sugar
+/// from `Fn<(A, B,), C>` to `Fn(A, B) -> C`
 pub(super) fn external_path(
     cx: &mut DocContext<'_>,
-    name: Symbol,
+    did: DefId,
     trait_did: Option<DefId>,
```

Comment:
`did` and `trait_did` will never be different if `trait_did.is_some()`, right? Could you change `trait_did: Option<DefId>` to `is_trait: bool` instead to avoid them getting out of sync?

Reply:
I did that, and then I did one better: It actually looks like `is_trait` is not even necessary. It is only used for sugaring `Fn` trait bounds, and rustdoc already checks that the `did` is a `Fn*` lang item.

Reply:
I made them as two separate commits, but I can squash them together if you would prefer.